pkg/apis/ocs/v1/storagecluster_types.go

```diff
@@ -106,6 +106,12 @@ type StorageDeviceSet struct {
 	// +optional
 	Replica int `json:"replica,omitempty"`
 
+	// DeviceType is the value of device type in
+	// this StorageDeviceSet. It can have only
+	// three values (SSD, HDD, NVMe)
+	// +optional
+	DeviceType string `json:"devicetype,omitempty"`
```

Comment:
If these are the only possible values we should be able to error out immediately if any other value is passed. Can you look into adding that?

Reply:
there can be case sensitive case. https://github.com/openshift/ocs-operator/pull/859#issuecomment-717880007